src/reference/02-DetailTopics/02-Configuration/15-In-Process-Classloaders.md

```diff
@@ -0,0 +1,92 @@
+all of the required classes in a lengthy list of jar files. Even with ssds, the
+io required to locate all of the classes and read them into memory can delay the
+startup of the task by seconds. To reduce this overhead, sbt creates layered
+classloaders where certain layers may be cached between task evaluation runs.
+The inital classloading overhead is a fixed cost, but the startup delay of
+subsequent runs in an interactive session can be reduced by caching the
+intermediate layers.
+
+Another benefit of the layered classloaders is that the jvm may choose to jit
+some of the cached classes. As a result, the total runtime of the task may
+decrease as the interactive session proceeds.
+
+
+### ClassLoader layering strategy
+As of 1.3.0, sbt allows users to select one of three classloader layering
+strategies[*]. The strategy may be set using the `classLoaderLayeringStrategy`
+setting. For example, to use the [AllLibraryJars] strategy in the test
+configuration, set
+
+```
+Test / classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.AllLibraryJars
+```
+
+Descriptions of the strategies follow.
+
+#### ScalaLibrary
+
+Prior to sbt 1.3.0, this was the only ClassLoader layering strategy. The top
+layer of this ClassLoader is able to load the scala standard library classes,
+the sbt test interface classes and the java library. This layer is cached
+between builds. The bottom layer of this classloader contains the rest of the
+project classpath which includes all of the library jar files as well as the
+project class files (or jars if `exportJars` is set). The bottom layer is never
+cached.
+
+Note that because the scala library is cached between runs, any shared global
+state created by the scala library is reused. In particular, the
+[global execution context](https://www.scala-lang.org/api/current/scala/concurrent/ExecutionContext\$.html)
+persists between runs. Avoid using the global execution context for tasks that
+run in process.
+
+The `ScalaLibrary` strategy is the default for all sbt 1.3.0 builds.
+
+#### AllLibraryJars
+
+Similar to `ScalaLibrary,` but adds an intermediate layer for the library
+dependencies of the project between the `ScalaLibrary` layer and the bottom
+project classpath layer. This strategy can significantly reduce the startup
+latency of run and test for projects with a large dependency classpath.
+
+Many projects will benefit in interactive sessions from switching to
+`AllLibraryJars`. The reason that it is not the default is because the caching
+of the dependency layer can cause errors if static fields are shared across
+runs. For example, some libraries use a global cache that is stored in a static
+field. If a project's `run` or `test` task depends on a particular initial
+state, they may fail on successive runs. Note that static global fields in the
+_project_ classpath do not cause this problem because the bottom layer is not
+cached between runs.
```

Comment:
I'd cut out the 

> For example, some libraries use a global cache that is stored in a static
field. If a project's `run` or `test` task depends on a particular initial
state, they may fail on successive runs. Note that static global fields in the
_project_ classpath do not cause this problem because the bottom layer is not
cached between runs.

since "because the caching of the dependency layer can cause errors if static fields are shared" is a good-enough summary.

Also project vs library could be difficult to distinguish if projects are split into multiple repos.